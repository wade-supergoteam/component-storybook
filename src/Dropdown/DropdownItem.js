import styled, { css } from "styled-components";

import { hoverFocus } from "../utils/hover";
import themeProp from "../utils/theme";

import {
  dropdownItemPaddingY,
  dropdownItemPaddingX,
  fontWeightNormal,
  dropdownLinkColor,
  dropdownLinkHoverColor,
  dropdownLinkHoverBg,
  dropdownLinkActiveColor,
  dropdownLinkActiveBg,
  dropdownLinkDisabledColor,
  enableGradients,
} from "./default-theme";

const DropdownItem = styled.a`
  display: block;
  width: 100%;
  padding: ${themeProp("dropdownItemPaddingY", dropdownItemPaddingY)} ${themeProp("dropdownItemPaddingX", dropdownItemPaddingX)};
  clear: both;
  font-weight: ${themeProp("fontWeightNormal", fontWeightNormal)};
  color: ${themeProp("dropdownLinkColor", dropdownLinkColor)};
  text-align: inherit;
  white-space: nowrap;
  background: none;
  border: 0;

  ${hoverFocus(css`
    color: ${themeProp("dropdownLinkHoverColor", dropdownLinkHoverColor)};
    text-decoration: none;
    background-color: ${themeProp("dropdownLinkHoverBg", dropdownLinkHoverBg)};
  `)};

  ${props =>
    props.active &&
    css`
      color: ${themeProp("dropdownLinkActiveColor", dropdownLinkActiveColor)}!important;
      text-decoration: none;
      background-color: ${themeProp("dropdownLinkActiveBg", dropdownLinkActiveBg)}!important;
    `};

  ${props =>
    props.disabled &&
    css`
      color: ${themeProp("dropdownLinkDisabledColor", dropdownLinkDisabledColor)}!important;
      background-color: transparent !important;

      ${themeProp("enableGradients", enableGradients) && "background-image: none;"};
    `};
`;

DropdownItem.defaultProps = {};

export default DropdownItem;
