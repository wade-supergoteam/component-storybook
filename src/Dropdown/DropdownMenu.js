import styled from "styled-components";

import themeProp from "../utils/theme";

import { borderRadius } from "../utils/border-radius";
import { boxShadow } from "../utils/box-shadow";

import {
  dropdownMinWidth,
  dropdownPaddingY,
  dropdownSpacer,
  fontSizeBase,
  bodyColor,
  dropdownBg,
  dropdownBorderWidth,
  dropdownBorderColor,
  dropdownBorderRadius,
  dropdownBoxShadow,
  zIndexDropdown,
} from "./default-theme";

const DropdownMenu = styled.div`
  position: absolute;
  top: 100%;
  z-index: ${themeProp("zIndexDropdown", zIndexDropdown)};
  display: none;
  float: left;
  min-width: ${themeProp("dropdownMinWidth", dropdownMinWidth)};
  padding: ${themeProp("dropdownPaddingY", dropdownPaddingY)} 0;
  margin: ${themeProp("dropdownSpacer", dropdownSpacer)} 0 0;
  font-size: ${themeProp("fontSizeBase", fontSizeBase)};
  color: ${themeProp("bodyColor", bodyColor)};
  text-align: left;
  list-style: none;
  background-color: ${themeProp("dropdownBg", dropdownBg)};
  background-clip: padding-box;
  border: ${themeProp("dropdownBorderWidth", dropdownBorderWidth)} solid ${themeProp("dropdownBorderColor", dropdownBorderColor)};
  ${borderRadius(themeProp("dropdownBorderRadius", dropdownBorderRadius))};
  ${boxShadow(themeProp("dropdownBoxShadow", dropdownBoxShadow))};
`;

DropdownMenu.defaultProps = {};

export default DropdownMenu;
