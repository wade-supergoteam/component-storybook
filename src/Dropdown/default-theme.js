// @flow

export const fontSizeBase = "1rem";
export const fontSizeSm = ".875rem";
export const fontWeightNormal = "400";
export const bodyColor = "#212529";
export const enableGradients = "false";

export const dropdownMinWidth = "10rem";
export const dropdownPaddingY = ".5rem";
export const dropdownSpacer = ".125rem";
export const dropdownBg = "#fff";
export const dropdownBorderColor = "rgba(0,0,0,0.15)";
export const dropdownBorderRadius = ".25rem";
export const dropdownBorderWidth = "1px";
export const dropdownDividerBg = "#e9ecef";
export const dropdownBoxShadow = "0 .5rem 1rem rgba(0,0,0,0.175)";
export const dropdownLinkColor = "#212529";
export const dropdownLinkHoverColor = "#16181b";
export const dropdownLinkHoverBg = "#f8f9fa";
export const dropdownLinkActiveColor = "#fff";
export const dropdownLinkActiveBg = "#007bff";
export const dropdownLinkDisabledColor = "#6c757d";
export const dropdownItemPaddingY = ".25rem";
export const dropdownItemPaddingX = "1.5rem";
export const dropdownHeaderColor = "#6c757d";
export const zIndexDropdown = 1000;
