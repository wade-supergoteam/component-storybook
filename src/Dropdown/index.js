export { default as DropdownHeader } from "./DropdownHeader";
export { default as DropdownItem } from "./DropdownItem";
export { default as DropdownMenu } from "./DropdownMenu";
export { default as DropdownDivider } from "./DropdownDivider";

export { default } from "./Dropdown";
