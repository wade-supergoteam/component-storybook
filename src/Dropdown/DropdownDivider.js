import styled from "styled-components";
import themeProp from "../utils/theme";

import { navDivider } from "../utils/nav-divider";

import { dropdownDividerBg } from "./default-theme";

const DropdownDivider = styled.div`
  ${navDivider(themeProp("dropdownDividerBg", dropdownDividerBg))};
`;

DropdownDivider.defaultProps = {};

export default DropdownDivider;
