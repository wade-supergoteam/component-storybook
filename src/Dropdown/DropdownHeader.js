import styled from "styled-components";

import themeProp from "../utils/theme";

import { dropdownPaddingY, dropdownItemPaddingX, fontSizeSm, dropdownHeaderColor } from "./default-theme";

import Heading from "../Heading";

const DropdownHeader = styled(Heading)`
  display: block;
  padding: ${themeProp("dropdownPaddingY", dropdownPaddingY)};
  ${themeProp("dropdownItemPaddingX", dropdownItemPaddingX)};
  margin-bottom: 0;
  font-size: ${themeProp("fontSizeSm", fontSizeSm)};
  color: ${themeProp("dropdownHeaderColor", dropdownHeaderColor)};
  white-space: nowrap;
`;

DropdownHeader.defaultProps = {};

export default DropdownHeader;
