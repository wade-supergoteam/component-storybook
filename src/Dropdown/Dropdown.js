import styled, { css } from "styled-components";

import DropdownHeader from "./DropdownHeader";
import DropdownItem from "./DropdownItem";
import DropdownMenu from "./DropdownMenu";
import DropdownDivider from "./DropdownDivider";

const Dropdown = styled.div`
  position: relative;

  ${props =>
    props.isOpen &&
    css`
      ${DropdownMenu} {
        display: block;
        ${props.align === "right" ? "right: 0;" : "left: 0;"};
      }
    `};
`;

Dropdown.Header = DropdownHeader;
Dropdown.Item = DropdownItem;
Dropdown.Menu = DropdownMenu;
Dropdown.Divider = DropdownDivider;

Dropdown.defaultProps = {
  align: "left",
};

export default Dropdown;
