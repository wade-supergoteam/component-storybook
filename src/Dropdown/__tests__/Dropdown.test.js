import React from "react";
import renderer from "react-test-renderer";
import Dropdown from "../";
import DropdownItem from "../DropdownItem";
import DropdownMenu from "../DropdownMenu";


test("create dropdown", () => {
    const wrapper = renderer.create(<Dropdown/>);
    expect(wrapper).toMatchSnapshot();
});

test("disabled isOpen", () => {
    const wrapper = renderer.create(<Dropdown isOpen={true}/>);
    expect(wrapper).toMatchSnapshot();
});

test("dropdown isOpen flase", () => {
    const wrapper = renderer.create(<Dropdown isOpen={false}/>);
    expect(wrapper).toMatchSnapshot();
});

test("dropdown with dropdownMenu left", () => {
    const wrapper = renderer.create(<Dropdown>
        <DropdownMenu align="left"/>
    </Dropdown>);
    expect(wrapper).toMatchSnapshot();
});

test("dropdown with dropdownMenu right", () => {
    const wrapper = renderer.create(<Dropdown>
        <DropdownMenu align="right"/>
    </Dropdown>);
    expect(wrapper).toMatchSnapshot();
});


test("active Dropdown", () => {
    const wrapper = renderer.create(<DropdownItem active/>);
    expect(wrapper).toMatchSnapshot();
});

test("disabled Dropdown", () => {
    const wrapper = renderer.create(<DropdownItem disabled/>);
    expect(wrapper).toMatchSnapshot();
});
