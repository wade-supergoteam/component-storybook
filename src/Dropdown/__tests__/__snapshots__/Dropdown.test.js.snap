// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`active Dropdown 1`] = `
.c0 {
  display: block;
  width: 100%;
  padding: .25rem 1.5rem;
  clear: both;
  font-weight: 400;
  color: #212529;
  text-align: inherit;
  white-space: nowrap;
  background: none;
  border: 0;
  color: #fff!important;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #007bff!important;
}

.c0:focus,
.c0:hover {
  color: #16181b;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #f8f9fa;
}

<a
  className="c0"
/>
`;

exports[`create dropdown 1`] = `
.c0 {
  position: relative;
}

<div
  className="c0"
/>
`;

exports[`disabled Dropdown 1`] = `
.c0 {
  display: block;
  width: 100%;
  padding: .25rem 1.5rem;
  clear: both;
  font-weight: 400;
  color: #212529;
  text-align: inherit;
  white-space: nowrap;
  background: none;
  border: 0;
  color: #6c757d!important;
  background-color: transparent !important;
  background-image: none;
}

.c0:focus,
.c0:hover {
  color: #16181b;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #f8f9fa;
}

<a
  className="c0"
  disabled={true}
/>
`;

exports[`disabled isOpen 1`] = `
.c0 {
  position: relative;
}

.c0 .c1 {
  display: block;
  left: 0;
}

<div
  className="c0"
/>
`;

exports[`dropdown isOpen flase 1`] = `
.c0 {
  position: relative;
}

<div
  className="c0"
/>
`;

exports[`dropdown with dropdownMenu left 1`] = `
.c0 {
  position: relative;
}

<div
  className="c0"
/>
`;

exports[`dropdown with dropdownMenu right 1`] = `
.c0 {
  position: relative;
}

.c0 .c1 {
  display: block;
  right: 0;
}

<div
  className="c0"
/>
`;
